core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -96,6 +96,11 @@
         Dynamic,
 
         /**
+         * mark this setting as final (the value cannot be updated)
```

Comment:
If we're looking to prevent updates when an index is closed then we might want to have a setting for that too. That way we can (eventually) require that all settings have one of the three.

I don't like not having one of these meaning "can only be updated on an index and only when that index is closed". If we have three update scenarios we may as well have three properties.